subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/fixture/AbstractJvmFailFastIntegrationSpec.groovy

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.testing.fixture
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.DefaultTestExecutionResult
+import org.gradle.integtests.fixtures.RepoScriptBlockUtil
+import org.gradle.test.fixtures.server.http.BlockingHttpServer
+import org.hamcrest.Matchers
+import org.junit.Rule
+
+abstract class AbstractJvmFailFastIntegrationSpec extends AbstractIntegrationSpec {
+    protected static final String FAILED_RESOURCE = "fail"
+    protected static final String OTHER_RESOURCE = "other"
+    protected static final int MAX_WORKERS = 2
+
+    @Rule
+    BlockingHttpServer server = new BlockingHttpServer()
+
+    def setup() {
+        server.start()
+    }
+
+    def "all tests run without fail fast"() {
```

Comment:
Since all these tests have similar structure, maybe you'd like to try [Data driven test](http://spockframework.org/spock/docs/1.1/data_driven_testing.html) with `where` clause. You can search `"where:"` in Gradle code base to see examples.

Reply:
Based on your comment, I started heading down this path.  Once I got to the test checking the interaction between `forkEvery` and `maxParallelForks`, the data table was getting a bit unwieldy.  In addition, with the recent commits, `TestNGFailFastIntegrationTest` now has a test verifying the interaction with `parallel` in TestNG.

If you feel strongly, let me know & I'll make another attempt to refactor these test methods to use `where:` clauses.